spec/aerospike/client_spec.rb

```diff
@@ -22,6 +22,21 @@
 
   let(:client) { Support.client }
 
+  [Aerospike::Replica::MASTER, Aerospike::Replica::MASTER_PROLES, Aerospike::Replica::SEQUENCE, Aerospike::Replica::RANDOM].each do |replica_policy|
+  context "alternate #replica_policies" do
+
+  before do
+    client.default_read_policy.replica = replica_policy
+    client.default_write_policy.replica = replica_policy
+    client.default_batch_policy.replica = replica_policy
```

Comment:
By running the entire client spec with different replica policies, you only verify that the client generally works with each policy. What you are not testing, is whether each policy actually works **as expected**. Some additional unit tests for `Cluster#master_node`, `Cluster#read_node`, and `Cluster#batch_read_node` would be good.

Reply:
I have verified them manually. Verifying them in tests require (partly external) infrastructure that does not exist for the Ruby client and is a project in itself.